src/ethereum/prague/blocks.py

```diff
@@ -103,3 +116,79 @@ class Receipt:
     cumulative_gas_used: Uint
     bloom: Bloom
     logs: Tuple[Log, ...]
+
+
+def validate_requests(requests: Tuple[Bytes, ...]) -> None:
```

Comment:
You should also verify that the request root doesn't contain spurious requests with an unknown type. Otherwise the block builder could stuff arbitrary data into the requests list which is obviously bad.

Probably EIP 7685 should be updated to explicitly say that requests may not be included unless authorised by another EIP.